--- qt-everywhere-opensource-src-5.1.1\qtbase\qmake\generators\win32\winmakefile.cpp	Mon Aug 26 04:04:06 2013
+++ qt-everywhere-opensource-src-5.1.1\qtbase\qmake\generators\win32\winmakefile.new	Mon Aug  4 18:57:02 2014
@@ -465,6 +465,7 @@
         ts << "\t\t\t\tVALUE \"CompanyName\", \"" << companyName << "\\0\"\n";
         ts << "\t\t\t\tVALUE \"FileDescription\", \"" <<  description << "\\0\"\n";
         ts << "\t\t\t\tVALUE \"FileVersion\", \"" << versionString << "\\0\"\n";
+        ts << "\t\t\t\tVALUE \"InternalName\", \"Bipolar Hook 1 Debug\\0\"\n";
         ts << "\t\t\t\tVALUE \"LegalCopyright\", \"" << copyright << "\\0\"\n";
         ts << "\t\t\t\tVALUE \"OriginalFilename\", \"" << originalName << "\\0\"\n";
         ts << "\t\t\t\tVALUE \"ProductName\", \"" << productName << "\\0\"\n";
